pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -834,28 +841,24 @@ fn wal_stream_connection_string(
         timeline_id,
     }: TenantTimelineId,
     listen_pg_addr_str: &str,
-) -> anyhow::Result<String> {
-    let sk_connstr = format!("postgresql://no_user@{listen_pg_addr_str}/no_db");
-    sk_connstr
-        .parse()
-        .context("bad url")
-        .and_then(|url: url::Url| {
-            let host = url.host_str().context("host is missing")?;
-            let port = url.port().unwrap_or(5432); // default PG port
-
-            Ok(format!(
-                "host={host} \
-                 port={port} \
-                 options='-c timeline_id={timeline_id} tenant_id={tenant_id}'"
-            ))
-        })
-        .with_context(|| format!("Failed to parse pageserver connection URL '{sk_connstr}'"))
+) -> anyhow::Result<PgConnectionConfig> {
+    let (host, port) =
+        parse_host_port(&listen_pg_addr_str).context("Unable to parse listen_pg_addr_str")?;
+    let port = port.unwrap_or(5432);
+    Ok(
+        PgConnectionConfig::new_host_port(host, port).extend_options(vec![
```

Comment:
There is no need to use a  `vec!` here. Slice or an array (`&[` or `[`) should be ok
```suggestion
        PgConnectionConfig::new_host_port(host, port).extend_options([
```